components/server/src/auth/resource-access.ts

```diff
@@ -152,6 +154,26 @@ export class CompositeResourceAccessGuard implements ResourceAccessGuard {
     }
 }
 
+/**
+ * FGAResourceAccessGuard can disable the delegate if FGA is enabled.
+ */
+export class FGAResourceAccessGuard implements ResourceAccessGuard {
+    constructor(private readonly userId: string, private readonly delegate: ResourceAccessGuard) {}
+
+    async canAccess(resource: GuardedResource, operation: ResourceAccessOp): Promise<boolean> {
```

Comment:
@svenefftinge Used your idea with the `ResourceGuard`, so headless-log is covered as well ([see below](https://github.com/gitpod-io/gitpod/pull/18519/files#diff-ad3bbabbf5bd475f8c522ec6cba0b0745ce8260a8989064f5feb6bb9e9a4b4d8R228)).